moonraker/components/td1.py

```diff
@@ -0,0 +1,337 @@
+                "error": self._error
+            }
+        }
+
+class TD1:
+    def __init__(self, config: ConfigHelper) -> None:
+        self.server = config.get_server()
+        self._config = config
+        self._loop = self.server.get_event_loop()
+        self.logger = logging.getLogger("td1")
+        self._server = config.get_server()
+        self._known_serials: set[Any]
+        self._vid: int = 0xE4B2
+        self._pid: int = 0x0045
+        self._enabled: bool = True
+        self._watch_task: Optional[asyncio.Task] = None
+        self._start_task: Optional[asyncio.Task] = None
+        self.serial_tasks: Dict = {}
+
+        self._register_endpoints()
+
+    def _register_endpoints(self) -> None:
+        self._server.register_endpoint(
+            "/machine/td1/data", RequestType.GET, self._handle_get_data
+        )
+        self._server.register_endpoint(
+            "/machine/td1/reboot", RequestType.POST, self._handle_reboot
+        )
+
+    async def close_device(self, device: TD1Protocol) -> None:
+        ret = device.close()
+        if ret is not None:
+            await ret
+
+    async def close(self) -> None:
+        # Stop watching for new devices
+        self._enabled = False
+        if self._start_task is not None:
+            await asyncio.wait_for(self._start_task, 1)
+
+        if self._watch_task is not None:
+            self._watch_task.cancel()
+            with contextlib.suppress(asyncio.CancelledError):
+                await asyncio.wait_for(self._watch_task, 5)
+
+        for td1 in self.serial_tasks.values():
+            await self.close_device(td1)
+
+    async def component_init(self) -> None:
+        self.logger.debug("Initializing TD1")
+        self._start_task = self._loop.create_task(self._start_all_serial_tasks())
+        self._watch_task = self._loop.create_task(self._watch_for_new_devices())
+
+    async def _start_all_serial_tasks(self) -> None:
+        devices = self._find_td1_devices()
+        if not devices:
+            self.logger.warning("No TD1 devices found.")
+            return
+        for port, serial_number in devices:
+            await self._start_serial_task(port, serial_number)
+            await asyncio.sleep(0.5)
+
+    async def _watch_for_new_devices(self) -> None:
+        self._known_serials = set(self.serial_tasks.keys())
+        while self._enabled:
+            await asyncio.sleep(5)  # Scan every 5 seconds
+
+            # Find currently connected devices
+            current_serials = set()
+            for port, serial_number in self._find_td1_devices():
+                current_serials.add(serial_number)
+                if serial_number not in self._known_serials:
+                    self.logger.info(f"Hot-plugged TD1 detected: {serial_number}")
+                    try:
+                        await self._start_serial_task(port, serial_number)
+                    except serial.serialutil.SerialException:
```

Comment:
There is probably no need for a `try` block here since no exception was handled.  Previously I believe that this was a bare exception and I indicated that `asyncio.CanceledError` needs to be re-raised.  So it would need to be something like:

```python
try:
    await self._start_serial_task(port, serial_number)
except asyncio.CancelledError:
    raise
except Exception:
    # log error here if you would like
    continue
```